lib/carbon/writer.py

```diff
@@ -150,13 +188,34 @@ def writeForever():
       writeCachedDataPoints()
     except Exception:
       log.err()
-      # Back-off on error to let time to the backend to recover.
+      # Back-off on error to give the backend time to recover.
       reactor.callLater(0.1, writeForever)
     else:
       # Avoid churning CPU when there are no metrics are in the cache
       reactor.callLater(1, writeForever)
 
 
+def writeTags():
+  while True:
+    tags = tagQueue.getbatch(settings.TAG_BATCH_SIZE)
+    if not tags:
+      break
+    state.database.tag(*tags)
+
+
+def writeTagsForever():
+  if reactor.running:
+    try:
+      written = writeTags()
```

Comment:
Maybe stupid question - why not just call `writeTags()`? Why assign to `written` and never check result? And writeTags() should not return anything, right?

Reply:
Yeah that was a by-product of some refactoring that I need to clean up